src/control/provider/system/system_linux.go

```diff
@@ -0,0 +1,111 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package system
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// DefaultProvider returns the package-default provider implementation.
+func DefaultProvider() *LinuxProvider {
+	return &LinuxProvider{}
+}
+
+// LinuxProvider encapsulates Linux-specific implementations of system
+// interfaces.
+type LinuxProvider struct{}
+
+// mountId,parentId,major:minor,root,mountPoint
+const (
+	_ int = iota
+	_
+	miMajorMinor
+	_
+	miMountPoint
+	miLastField
+)
+
+func scanMountInfo(input io.Reader, target string, scanField int) (bool, error) {
+	scn := bufio.NewScanner(input)
+	for scn.Scan() {
+		fields := strings.Fields(scn.Text())
+		if len(fields) < miLastField {
+			continue
+		}
+		if fields[scanField] == target {
+			return true, nil
+		}
+	}
+
+	return false, scn.Err()
+}
+
+// IsMounted checks the target device or directory for mountedness.
+func (s LinuxProvider) IsMounted(target string) (bool, error) {
+	st, err := os.Stat(target)
+	if err != nil {
+		return false, err
+	}
+
+	var scanField int
+	switch {
+	case st.IsDir():
+		scanField = miMountPoint
+	case st.Mode()&os.ModeDevice != 0:
```

Comment:
I'm not clear on what the syntax is doing, could you please elaborate?

Reply:
It's a bitwise comparison. The stat's mode bits are ANDed with the os.ModeDevice bitmask. If the result is non-zero, the target is a device, and we want to look for the device's major:minor to see if it's already mounted somewhere. I guess the Go authors didn't provide an IsDevice() function because not all OSes support the concept of unix-style devices.

Reply:
ah, I think I was confused because there weren't spaces surrounding the "&", thanks